pkg/util/testutil/timeout.go

```diff
@@ -13,6 +13,18 @@ import (
 // Condition is evaluated in a goroutine to avoid tests hanging if a system
 // is deadlocked.
 func AssertTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) {
+	internalTrueBeforeTimeout(t, false, frequency, timeout, condition)
+}
+
+// RequireTrueBeforeTimeout is the same as AssertTrueBeforeTimeout, but it calls
+// t.failNow() if the condition function times out.
+func RequireTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) {
+	if !internalTrueBeforeTimeout(t, true, frequency, timeout, condition) {
+		t.FailNow()
+	}
+}
+
+func internalTrueBeforeTimeout(t *testing.T, require bool, frequency, timeout time.Duration, condition func() bool) bool {
```

Comment:
nit: I think it's clearer if the "new" argument is passed at the end, i.e.:

```suggestion
func internalTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool, require bool) {
```

(also removed the return value, related to my comment above)

Reply:
The require arguments is actually unused, it was related to a previous version of my PR but then I forgot to remove it after an update. I'll just remove it altogether.